src/Controls/src/Xaml/XamlNode.cs

```diff
@@ -151,7 +151,7 @@ public override void Accept(IXamlNodeVisitor visitor, INode parentNode)
 			{
 				foreach (var node in Properties.Values.ToList())
 					node.Accept(visitor, this);
-				foreach (var node in CollectionItems)
+				foreach (var node in CollectionItems.ToList())
```

Comment:
It might be good to add a comment here saying why this is needed.

(minor) If it is just to copy the contents so the `foreach` doesn't fail if elements are added/removed, it is usually better to just call `ToArray()` instead.